src/ui/public/config/config.js

```diff
@@ -129,34 +35,84 @@ module.service('config', function (Private, kbnVersion, kbnIndex, $rootScope, bu
    *                             be stored. Defaults to the config key
    * @return {function} - an unbind function
    */
-  config.$bind = function ($scope, key, property) {
-    if (!property) property = key;
-
-    let update = function () {
-      $scope[property] = config.get(key);
-    };
+  config.$bind = function (scope, key, property = key) {
+    config.on(key, update, scope);
+    function update(value) {
+      scope[property] = value;
+    }
+  };
 
+  config.on = function (key, fn, scope = $rootScope) {
+    const postfix = key === `*` ? `` : `.${key}`;
```

Comment:
Not really? I think right now you just do `config.set(key, value)` and if `key` doesn't have an entry in the `settings/../defaults.js` file then its assumed to be a custom property.
For that reason people would need to always have a defined value for a custom property _(via `config.set`)_ before they register a listener. (Which may be an acceptable constraint to have, I guess, just have them initialize it and provide a meaningful error message on misses)